static/app/views/settings/components/forms/controls/rangeSlider.tsx

```diff
@@ -46,160 +48,159 @@ type Props = {
+    return allowedValues[newSliderValue];
+  }
 
-  handleBlur = e => {
-    const {onBlur} = this.props;
+  function handleInput(e: ChangeEvent<HTMLInputElement>) {
+    const newSliderValue = parseInt(e.target.value, 10);
+    setSliderValue(newSliderValue);
+    onChange?.(getActualValue(newSliderValue), e);
+  }
+
+  function handleCustomInputChange(e: ChangeEvent<HTMLInputElement>) {
+    const newSliderValue = parseInt(e.target.value, 10);
+    setSliderValue(isNaN(newSliderValue) ? 0 : value);
+  }
+
+  function handleBlur(e: MouseEvent<HTMLInputElement> | KeyboardEvent<HTMLInputElement>) {
     if (typeof onBlur !== 'function') {
       return;
     }
 
     onBlur(e);
-  };
-
-  handleCustomInputChange = e => {
-    const value = parseInt(e.target.value, 10);
-    this.setValue(isNaN(value) ? 0 : value);
-  };
-
-  handleCustomInputBlur = e => {
-    this.handleInput(e);
-  };
-
-  render() {
-    let {min, max, step} = this.props;
-    const {
-      name,
-      disabled,
-      allowedValues,
-      formatLabel,
-      placeholder,
-      showCustomInput,
-      className,
-    } = this.props;
-    const {sliderValue} = this.state;
-    let actualValue = sliderValue;
-    let displayValue: React.ReactNode = actualValue;
-
-    if (allowedValues) {
-      step = 1;
-      min = 0;
-      max = allowedValues.length - 1;
-      actualValue = allowedValues[sliderValue];
-      displayValue =
-        typeof actualValue !== 'undefined' ? actualValue : t('Invalid value');
+  }
+
+  function getSliderData() {
+    if (!allowedValues) {
+      const {min, max, step} = props;
+      return {
+        min,
+        max,
+        step,
+        actualValue: sliderValue,
+        displayValue: sliderValue,
+      };
     }
 
-    displayValue =
-      typeof formatLabel === 'function' ? formatLabel(actualValue) : displayValue;
-
-    return (
-      <div className={className}>
-        {!showCustomInput && <Label htmlFor={name}>{displayValue}</Label>}
-        <SliderAndInputWrapper showCustomInput={showCustomInput}>
-          <Slider
-            type="range"
-            name={name}
-            min={min}
-            max={max}
-            step={step}
-            disabled={disabled}
-            onInput={this.handleInput}
-            onChange={() => {}}
-            onMouseUp={this.handleBlur}
-            onKeyUp={this.handleBlur}
+    const actualValue = allowedValues[sliderValue];
+
+    return {
+      step: 1,
+      min: 0,
+      max: allowedValues.length - 1,
+      actualValue,
+      displayValue: defined(actualValue) ? actualValue : t('Invalid value'),
+    };
+  }
+
+  const {min, max, step, actualValue, displayValue} = getSliderData();
+
+  return (
+    <div className={className} ref={forwardRef}>
+      {!showCustomInput && (
+        <Label htmlFor={name}>
+          {typeof formatLabel === 'function' ? formatLabel(actualValue) : displayValue}
```

Comment:
How about `formatLabel?.(actualValue) ?? displayValue`

Reply:
sure, but this was also copied

Reply:
oof sorry haha, I know I'm a little hypocritical here. Sometimes I fix stuff like this when I refactor, other times I don't

Reply:
It's a difficult diff to review ;-)

Reply:
I updated it 😉